pkg/worker/stage/stage.go

```diff
@@ -78,7 +81,32 @@ func (sm *stageManager) SetRecordInfo(project, pipeline, recordID string) {
 	sm.recordID = recordID
 }
 
-func (sm *stageManager) ExecCodeCheckout(token string, stage *api.CodeCheckoutStage) error {
+func (sm *stageManager) SetEvent(e *api.Event) {
```

Comment:
Why need this function? We can directly set the event.

Reply:
Authority control，other package can not get the event(variable e).